stetho/src/main/java/com/facebook/stetho/inspector/protocol/module/DOM.java

```diff
@@ -172,64 +208,196 @@ public void run() {
     });
   }
 
-  private Node createNodeForElement(Object element) {
+  private void updateTree() {
+    long startTimeMs = SystemClock.elapsedRealtime();
+
+    ShadowDOM.Update domUpdate = createShadowDOMUpdate();
+    boolean isEmpty = domUpdate.isEmpty();
+    if (isEmpty) {
+      domUpdate.abandon();
+    } else {
+      applyDOMUpdate(domUpdate);
+    }
+
+    long endTimeMs = SystemClock.elapsedRealtime();
+    long deltaMs = endTimeMs - startTimeMs;
+    LogUtil.e(String.format(
+        "DOM.updateTree() completed in %s ms%s",
+        Long.toString(deltaMs),
```

Comment:
A few comments here:

1. Why use `Long.toString` when you can use use `%d`?  Java doesn't have any of that `%ld` stuff :)
2. `LogUtil.e` already supports formatting, so the extra `String.format` is unnecessary.
3. `LogUtil.e` is _way_ too verbose :)
4. Technically if you use `String.format` in debug code you should always write `String.format(Locale.US, ...`.  It's best not to assume you know how localization works even in this case it is safe.